### PR TITLE
Skip binning prep when no binning/aDNA

### PR DESCRIPTION
**Pull request by jfy133**
As reported by @prototaxites on [slack](https://nfcore.slack.com/archives/CE9MS66BS/p1688544402597859)

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#439](https://github.com/nf-core/mag/pull/445) - Fix bug in assembly input (by @prototaxites)
 - [#447](https://github.com/nf-core/mag/pull/447) - Remove `default: None` from parameter schema (by @drpatelh)
 - [#449](https://github.com/nf-core/mag/pull/447) - Fix results file overwriting in Ancient DNA workflow (reported by @alexhbnr, fix by @jfy133)
+- [#XXX](https://github.com/nf-core/mag/pull/447) - Fix binning preparation from running even when binning was requested to be skipped (reported by @prototaxites, fix by @jfy133)
```
Comment (jfy133) [inline CHANGELOG.md:34]:
```suggestion
- [#470](https://github.com/nf-core/mag/pull/470) - Fix binning preparation from running even when binning was requested to be skipped (reported by @prototaxites, fix by @jfy133)
```

File: workflows/mag.nf [hunk 2]
```diff
@@ -617,10 +617,12 @@ workflow MAG {
     ch_busco_summary            = Channel.empty()
     ch_checkm_summary           = Channel.empty()
 
-    BINNING_PREPARATION (
-        ch_assemblies,
-        ch_short_reads
-    )
+    if ( !skip_binning || params.ancient_dna ) {
```
Comment (prototaxites) [inline workflows/mag.nf:620]: `params.skip_binning`*

**Comment by prototaxites** [comment]
With those changes, binning entry mostly seems to run fine - but I'm getting a weird error with Centrifuge that's only occurring sporadically:

```
-[nf-core/mag] Pipeline completed with errors-
ERROR ~ Error executing process > 'NFCORE_MAG:MAG:CENTRIFUGE (test_minigut-minigut_cf)'

Caused by:
  java.util.NoSuchElementException -- Check script './workflows/../modules/local/centrifuge.nf' at line: 21

Source block:
  def input = meta.single_end ? "-U \"${reads}\"" :  "-1 \"${reads[0]}\" -2 \"${reads[1]}\""
  prefix = task.ext.prefix ?: "${meta.id}"
  """
      centrifuge -x "${db_name}" \
          -p ${task.cpus} \
          --report-file report.txt \
          -S results.txt \
          $input
      centrifuge-kreport -x "${db_name}" results.txt > ${prefix}.centrifuge_kreport.txt
      cat results.txt | cut -f 1,3 > results.krona
  
      cat <<-END_VERSIONS > versions.yml
      "${task.process}":
          centrifuge: \$(centrifuge --version | sed -n 1p | sed 's/^.*centrifuge-class version //')
      END_VERSIONS
      """

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`

 -- Check '.nextflow.log' file for details
```

I can't figure out a way to reliably reproduce it, and foolishly lost `.nextflow.log` somewhere in my attempts to do so. Line 21 is the one that defines `$prefix` - perhaps there is some kind of race condition where if centrifuge at the wrong time, something has modified the meta map in an unsafe way?

File: workflows/mag.nf (hunk 2 above)
Comment (jfy133) [reply workflows/mag.nf]: Should have added everywhere!

File: workflows/mag.nf [hunk 3]
```diff
@@ -688,8 +688,10 @@ workflow MAG {
                 }
         }
 
-        ch_versions = ch_versions.mix(BINNING_PREPARATION.out.bowtie2_version.first())
-        ch_versions = ch_versions.mix(BINNING.out.versions)
+        if ( !params.skip_binning || params.ancient_dna ) {
```
Comment (prototaxites) [inline workflows/mag.nf:691]: If we skip binning but enable ancient DNA - will this fail as BINNING.out doesn't exist?

Comment (jfy133) [reply workflows/mag.nf:691]: Ah yeah, sorry I shouldn't be doing two pipelines at once (I thought this wouldve been a easier fix than I thought 😅)

I'll come back to this tomorrow morning in my normal mag Dev time

Thanks for the rapid feedback though, it's appreciated!

Comment (prototaxites) [reply workflows/mag.nf:691]: No worries, it's just me procrastinating on what I should be doing... 😅

Comment (jfy133) [reply workflows/mag.nf]: OK indeed- it tripped me up the binning prep version mixing wasn't in the same place (same with binning) as the actual invocation of the module!

Comment (jfy133) [reply workflows/mag.nf]: Should be good now: binning prep versions mixed in when executed, and binning is always mixed in as it's already in the binning activiation if/else statement

**Comment by jfy133** [comment]
@prototaxites failure is linting which will be resolved in #467

**Review by prototaxites — APPROVED** [review]
LGTM!

**Pull request merged**